external/storm-hdfs/src/main/java/org/apache/storm/hdfs/trident/HdfsState.java

```diff
@@ -136,33 +193,40 @@ public void run() {
         private transient FSDataOutputStream out;
         protected RecordFormat format;
         private long offset = 0;
+        private int bufferSize =  131072; // default 128 K
 
-        public HdfsFileOptions withFsUrl(String fsUrl){
+        public HdfsFileOptions withFsUrl(String fsUrl) {
             this.fsUrl = fsUrl;
             return this;
         }
 
-        public HdfsFileOptions withConfigKey(String configKey){
+        public HdfsFileOptions withConfigKey(String configKey) {
             this.configKey = configKey;
             return this;
         }
 
-        public HdfsFileOptions withFileNameFormat(FileNameFormat fileNameFormat){
+        public HdfsFileOptions withFileNameFormat(FileNameFormat fileNameFormat) {
             this.fileNameFormat = fileNameFormat;
             return this;
         }
 
-        public HdfsFileOptions withRecordFormat(RecordFormat format){
+        public HdfsFileOptions withRecordFormat(RecordFormat format) {
             this.format = format;
             return this;
         }
 
-        public HdfsFileOptions withRotationPolicy(FileRotationPolicy rotationPolicy){
+        public HdfsFileOptions withRotationPolicy(FileRotationPolicy rotationPolicy) {
             this.rotationPolicy = rotationPolicy;
             return this;
         }
 
-        public HdfsFileOptions addRotationAction(RotationAction action){
+        public HdfsFileOptions withBufferSize(int size) {
+            this.bufferSize = Math.max(4096, size); // at least 4K
```

Comment:
The default is 131072 (128 K). Here just ensuring that, if at all user tries to change the default, it does not go below 4 K.